toolkit/tools/grapher/grapher.go

```diff
@@ -267,10 +302,8 @@ func populateGraph(graph *pkggraph.PkgGraph, repo *pkgjson.PackageRepo) (err err
 
 	// Scan and add each package we know about
 	logger.Log.Infof("Adding all packages from %s", *input)
-	// NOTE: range iterates by value, not reference. Manually access slice
-	for idx := range packages {
-		pkg := packages[idx]
-		err = addLocalPackage(graph, pkg)
+	for _, pkg := range packages {
```

Comment:
I don't recall exactly what was going wrong, but if I recall right it ended up being that having two different versions of the `pkg` object floating around with different pointers ended up causing a bunch of headaches (I think it because we want to operate on the original copies of the packages in the repo object, not clones of them?)

Reply:
I'm not sure I follow. The `pkg` variable is a pointer regardless of the way we initiate it here - we're still operating on the same object. As far as I could, I also believe that both ways of initialization are equivalent in terms of result - no cloning is happening anywhere.

Reply:
Maybe this is from before we had everything defined as pointers... seems ok looking again.